### ISSUE TITLE
Keyboard focus issues 1.8 thru 1.10

### ISSUE THREAD

**Issue by qgates** [mate-desktop/marco#212]
Some longstanding focus issues seem to be plaguing MATE that are not apparent in other desktops (eg. Cinnamon/XFCE). Test platforms include Linux Mint 17.0/17.1/17.2 (old install, Toshiba laptop, RV250 gfx), 17.2 Virtual box (fresh install) and LMDE 2 fresh install (MATE 1.10, Lenovo T430 laptop). As a mainly keyboard user, this makes MATE frustrating at present.

Some instances of this issue are sporadic, ie. come and go, where others are repeatable. For sporadic examples, it's difficult to pinpoint the circumstances under which the issue arises, but once present the issue remains until the user switches to a physical tty (Ctrl+Alt+1) and back to the desktop, or enters and resumes from suspend.

**Example**

Ctrl+Alt+Del to bring up shutdown dialog. Dialog has no keyboard focus ie. Alt+U to suspend won't work. Mouseclick on window restores focus allowing keyboard shortcuts to work. No focus switching keys (Alt+Tab, Alt+Esc etc) will focus the window. When MATE is in this "state" a number of other popup dialogs and spawned windows won't receive keyboard focus either.

Interestingly, and possibly a clue to what's going on, when in this state the Clementine music player global keyboard shortcuts double up; ie. pressing hotkey to toggle playback pause/resume causes clementine to pause and immediately resume in succession (the effect of pressing the key twice).

After resuming from suspend the problem is cleared for a while, until it invariably returns.

Since this issue is sporadic ie. doesn't happen all the time and only under certain conditions, perhaps we can use this thread to gather more info and evidence towards a fix. It's my biggest frustration with MATE presently.

There are other less sporadic focus issues described elsewhere which may be related eg. #208 #117.

**Comment by monsta** [comment]
Thanks for the detailed analysis. I didn't even stumble upon these problems except for what I've reported in https://github.com/mate-desktop/marco/issues/117.
If you have some time, could you please check if all that happens with Metacity as well?

**Comment by ToApolytoXaos** [comment]
I have the same issue whenever I close a window and press `Alt-F2` to run a certain command I need; this is really annoying and I had to switch back to XFCE with the hope MATE would have resolved this problem with the release of 1.10.

Apparently it has not and it's a sad thing; I hate using the mouse by clicking anywhere on the desktop in order to regain keyboard focus.

**Comment by zaplo00** [comment]
Not sure if this is caused by same issue but switching some wine games fullscreen  causes keyboard and mouse clicks to stop working until killing the program. Mouse cursor moves and tooltips show (e.g. in clock), but it's impossibly to click anywhere and mouse movement is limited to first monitor as if it's in fullscreen when desktop shows. 

This can only be fixed by Ctrl-Alt-Fx terminal by killing the program. This behavior happens with marco in mate, i don't have same problem in other desktops.
